GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -1484,7 +1484,6 @@ void DoCommit()
 
                     ScriptManager.RunEventScripts(this, ScriptEvent.AfterCommit);
 
-                    Message.Text = string.Empty; // Message.Text has been used and stored
                     _commitMessageManager.ResetCommitMessage();
                     CommitKind = CommitKind.Normal;
```

Comment:
These three lines belong together.

Reply:
Why? I think, this is exactly the place where the commit message text box is cleared.

Reply:
I reverted my last commit, after you answer this question, will do a new commit.

Reply:
I have checked it again, and, I think, the empty message box statement should be moved a couple line below

Reply:
These three lines reset the commit message and whether it is to be stored or not (refer to `CommitKind` and the related change history). That's why I do not want to tear them apart.

You are right with https://github.com/gitextensions/gitextensions/pull/8513#issuecomment-703305389! 👍 
Moving `Message.Text = string.Empty` would result in the commit message wrongly be kept on exception.
Not critical but not correct. So nothing should be changed here at all.

Reply:
> So nothing should be changed here at all.

Or else
```
// Message.Text has been used and stored
_commitMessageManager.ResetCommitMessage();
CommitKind = CommitKind.Normal;
bool pushCompleted = true;
try
{
    if (push) //...
}
finally
{
    Message.Text = string.Empty;
}
```

Reply:
I'd go with @mstv suggestion - this way we can be sure the text gets cleared.

Reply:
With this `try {} finally {}` we can solve the issue without much effort, i.e. display the message of what is being pushed (and safely reset it right after the push).
Restore the branch in GH and reopen this PR, please.

Reply:
Done!